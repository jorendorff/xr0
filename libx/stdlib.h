#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [
	if (@ptr)
		.dealloc ptr;
	else /* calling free on a non-allocated pointer */
		.undefined;
];

#endif
