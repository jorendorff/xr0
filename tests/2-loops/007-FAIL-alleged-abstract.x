#include <stdlib.h>

void
unit()
{
	int i;
	void **p;

	p = malloc(9);

	for (i = 0; i < 9; i++) [ .alloc p[i]; ] {}

	/* should fail loop internal consistency check */
}
