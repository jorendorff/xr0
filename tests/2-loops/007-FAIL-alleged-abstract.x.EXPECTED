cannot exec statement: qed error: actual and alleged states differ
